kernel.cpp

```diff
@@ -57,6 +57,17 @@ static long kbdMatrixKeyCodes[8][8] = {
  {KEYCODE_F5,         KEYCODE_e,         KEYCODE_t, KEYCODE_u, KEYCODE_o, KEYCODE_LeftBracket,  KEYCODE_Equals,       KEYCODE_q},
  {KEYCODE_Insert,     KEYCODE_LeftShift, KEYCODE_x, KEYCODE_v, KEYCODE_n, KEYCODE_Comma,        KEYCODE_Slash,        KEYCODE_Escape},
 };
+#else if defined(RASPI_C128)
+static long kbdMatrixKeyCodes[11][8] = {
```

Comment:
These dimensions are backwards from the data provided.  Also, if the dimension is going to change then the kbdPA iterator later on in this file would have to change to 0->10 but it is still only 0-7.  Then the gpioPins array would also have to be adjusted for the scan to actually include the extra 3 pins. gpioPins[8] and gpioPins[9] and gpioPins[10] would have to become the three new PA lines.  Then the remaining would have to shift up.  I don't think this change is sufficient to do what you want it to do.  There would be more work elsewhere and a bunch of conditionals to extend the PA lines only for the C128 builds.

Reply:
Thanks ever so much for looking.  I will try and fix based on your comments.